ragna/deploy/_ui/main_page.py

```diff
@@ -41,26 +41,28 @@ def __init__(self, api_wrapper, template):
         )
 
     async def refresh_data(self):
-        self.chats = await self.api_wrapper.get_chats()
-        self.components = self.api_wrapper.get_components()
-        self.corpus_metadata = await self.api_wrapper.get_corpus_metadata()
-        self.corpus_names = await self.api_wrapper.get_corpus_names()
+        self.chats = self._engine.get_chats(user=pn.state.user)
+        self.components = self._engine.get_components()
+        self.corpus_metadata = await self._engine.get_corpus_metadata()
+        self.corpus_names = await self._engine.get_corpuses()
 
     @param.depends("chats", watch=True)
     def after_update_chats(self):
         self.left_sidebar.chats = self.chats
 
         if len(self.chats) > 0:
             chat_id_exist = (
-                len([c["id"] for c in self.chats if c["id"] == self.current_chat_id])
+                len(
```

Comment:
Nits:

1. I don't see a need to turn the chat IDs into strings here. We should be able to just use the `UUID` directly without all the conversion.
2. The expression is pretty convoluted and inefficient. If you do touch this line again, we should do something like

    ```suggestion
                chat_id_exists = any(c.id == self.current_chat_id for c in self.chats)
    ```

    That avoids creating a list that we know can only have 0 or 1 elements and terminates immediately when a match is found instead of going through all chats. Plus, and this is the main point here, it is even simpler.

Reply:
Feel free to address in this PR or a follow up.

Reply:
Fixed